src/algorithms/MyArrayList.java

```diff
@@ -0,0 +1,151 @@
+package src.algorithms;
+
+import java.io.Serializable;
+import java.util.*;
+
+public class MyArrayList<T>  implements Iterable<T>, Serializable {
+    private static final long serialVersionUID = 1L;
+
+    private static final int DEFAULT_CAPACITY = 10;
+    transient Object[] elements;
+    private int size;
+
+    public MyArrayList() {
+        this.elements = new Object[DEFAULT_CAPACITY];
+    }
+
+    public MyArrayList(int initialCapacity) {
+        if (initialCapacity >= 0) {
+            this.elements = new Object[initialCapacity];
+        } else {
+            throw new IllegalStateException("Начальная емкость (initialCapacity) не может быть меньше или равен нулю");
+        }
+    }
+    public int size() {
+        return size;
+    }
+
+    public boolean isEmpty() {
+        return size == 0;
+    }
+
+    private Object[] capacityGrowth(int minCapacity ) {
+        int capacity = elements.length;
+        if (capacity > 0) {
+            Object[] newCapacity = new Object[(elements.length * 2)];
+            System.arraycopy(elements, 0, newCapacity, 0, elements.length);
+            return newCapacity;
+        }else{
+            return elements = new Object[Math.max(DEFAULT_CAPACITY, minCapacity)];
+        }}
+
+    private Object[] capacityGrowth() {
+        return capacityGrowth(size + 1);
+    }
+
+    public boolean add(T object){
+        if (size == elements.length){
+            elements = capacityGrowth();
+        }
+        elements[size] = object;
+        size++;
+        return true;
+    }
+
+    public boolean addAll(MyArrayList otherList) {
+        if (otherList == null || otherList.size == 0) {
+            return false;
+        }
+
+        int newSize = this.size + otherList.size;
+
+        if (newSize > this.elements.length) {
+            this.elements = capacityGrowth(newSize);
+        }
+
+        if (otherList.elements != null) {
+            System.arraycopy(otherList.elements, 0, this.elements, this.size, otherList.size);
+            this.size = newSize;
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public void addAll(List<Object> list) {
+        for (Object obj : list) {
+            this.add((T) obj);
+        }
+    }
+
+    private int checkingIndex(int index){
+        if(index > size || index < 0){
```

Comment:
Кароче приведи все к единому кодстайлу. У тебя в половине случаев есть пробел перед фигурной скобкой, а в половине - нет
```suggestion
        if(index > size || index < 0) {
```